brave/src/test/java/brave/internal/baggage/UnsafeArrayMapTest.java

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.internal.baggage;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import org.assertj.core.api.AbstractMapAssert;
+import org.junit.Test;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.entry;
+
+public class UnsafeArrayMapTest {
```

Comment:
even though  UnsafeArrayMap is package private and only used to implement baggage fields. I kept it a generic map type as it makes testing easier. In other words, that this takes arbitrary generic type params != a plan to expose this. Mostly UnsafeArrayMap allows us to safely share views of all baggage state without copying arrays or values into new map.